src/Areas/Server/Commands/ToolLoading/ServerToolLoader.cs

```diff
@@ -480,7 +481,7 @@ await request.Server.NotifyProgressAsync(progressToken.Value,
                 }
                 if (root.TryGetProperty("parameters", out var paramsProp) && paramsProp.ValueKind == JsonValueKind.Object)
                 {
-                    parameters = JsonSerializer.Deserialize(paramsProp.GetRawText(), ServerJsonContext.Default.DictionaryStringObject) ?? [];
+                    parameters = paramsProp.EnumerateObject().ToDictionary(prop => prop.Name, prop => (object?)prop.Value) ?? [];
```

Comment:
this is a simple iteration without out any deserialization overhead, and with these changes we also drop need for `DictionaryStringObject `